core/sql/executor/OrcFileReader.cpp

```diff
@@ -122,24 +150,28 @@ OFR_RetCode OrcFileReader::init()
 OFR_RetCode OrcFileReader::open(const char* path)
 {
   QRLogger::log(CAT_SQL_HDFS_ORC_FILE_READER, LL_DEBUG, "OrcFileReader::open(%s) called.", path);
+
+  if (initJNIEnv() != JOI_OK)
+     return OFR_ERROR_OPEN_PARAM;
   jstring js_path = jenv_->NewStringUTF(path);
   if (js_path == NULL) 
+  {
+    jenv_->PopLocalFrame(NULL);
```

Comment:
I'm a little confused about the rules for when a call to PopLocalFrame is needed. Is it needed whenever we have previously done a call into any jenv_ method? (in this case NewStringUTF).

Reply:
PopLocalFrame should be called before returning from the method after initJNIEnv() is called.

Reply:
Thanks.